Emulated hue component now correctly reports non-dimmable entities

sigwinch28 opened the pull request:
## Description:

This is required as 3rd generation echo dot devices will first
attempt to turn a device to full brightness *and* turn it on if
it is seen as a dimmable light.

Instead, we should report non-dimmable devices as on/off only,
meaning that the echo dot will only attempt to turn the device on
and off, instead of dimming it.

**Related issue (if applicable):** fixes #13296

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.

sigwinch28 committed:
Emulated hue component now correctly reports non-dimmable entities

Fixes #13296.

This is required as 3rd generation echo dot devices will first
attempt to turn a device to full brightness *and* turn it on if
it is seen as a dimmable light.

Instead, we should report non-dimmable devices as on/off only,
meaning that the echo dot will only attempt to turn the device on
and off, instead of dimming it.

homeassistant commented:
Hi @sigwinch28, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#18749).

Once you do that we will be able to review and accept this pull request.

Thanks!

sigwinch28 committed:
adhere to PEP8

balloob reviewed (changes requested):
Can you add a test to make sure it works

sigwinch28 committed:
Add tests for emulated_hue light types.

This required an extension to the demo light platform in order to
create a non-dimmable light depending on the value of the optional
configuration value 'demo_with_basic' (which is a boolean, default
False).

Furthermore, had to update a configuration-checking test to
deal with the new light demo platform optional config value.

sigwinch28 committed:
PEP8 fix

sigwinch28 committed:
fix docstrings in basic demo light

balloob commented on homeassistant/components/light/demo.py:
```diff
@@ -23,17 +26,27 @@
 SUPPORT_DEMO = (SUPPORT_BRIGHTNESS | SUPPORT_COLOR_TEMP | SUPPORT_EFFECT |
                 SUPPORT_COLOR | SUPPORT_WHITE_VALUE)
 
+CONF_WITH_BASIC = 'demo_with_basic'
```
Please don't add any config options to our demo platforms, especially to satisfy a test.

sigwinch28 commented:
Okay, how should I write a test to confirm it works?

All of the demo lights are dimmable, so I can't use one of them to check for the correct API response. Should I be manually injecting an entity somehow?

dgomes commented:
Just create a MockBasicLight in your test

sigwinch28 commented:
I would, but I can't seem to find any examples of `MockBasicLight`, or examples of adding mock entities during testing. Is there any thorough test infrastructure documentation?

dgomes commented:
In the code you are touching (test_hue) you can find several examples from a mock bridge to very simple mock lights (example test_available)

balloob commented:
This PR seems to have gone stale. Closing it.
